### PR TITLE
weechatScripts.weechat-matrix: unstable-2019-11-10 -> unstable-2020-01-21

### PR DESCRIPTION
**Pull request by emilazy**
###### Motivation for this change

Updated the sources and dependencies for matrix-nio and its dependent weechat-matrix. Not tested extensively, but "it builds".

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by emilazy** [comment]
@GrahamcOfBorg build weechatScripts.weechat-matrix

**Comment by emilazy** [comment]
@GrahamcOfBorg build weechatScripts.weechat-matrix

**Comment by emilazy** [comment]
@GrahamcOfBorg build weechatScripts.weechat-matrix

**Comment by emilazy** [comment]
Updated to include the new `matrix_{decrypt,sso_helper}` scripts.

@GrahamcOfBorg build weechatScripts.weechat-matrix

**Comment by emilazy** [comment]
Missed a dependency for the SSO helper that wasn't actually affecting anything because it was coming in indirectly via `requests`... Python packaging is a mess.

@GrahamcOfBorg build weechatScripts.weechat-matrix

cc @tilpner

File: pkgs/applications/networking/irc/weechat/scripts/weechat-matrix/default.nix [hunk 1]
```diff
@@ -38,10 +53,18 @@ in buildPythonPackage {
...
+      $src/contrib/matrix_sso_helper \
+      $out/bin/
     substituteInPlace $out/bin/matrix_upload \
-      --replace '/usr/bin/env -S python3 -u' '${matrixUploadPython}/bin/python -u' 
-  
+      --replace '/usr/bin/env -S python3' '${scriptPython}/bin/python'
+    substituteInPlace $out/bin/matrix_sso_helper \
+      --replace '/usr/bin/env -S python3' '${scriptPython}/bin/python'
+    substituteInPlace $out/bin/matrix_decrypt \
+      --replace '/usr/bin/env python3' '${scriptPython}/bin/python'
```
[the remark this answers is not in the document]
Comment (emilazy) [reply pkgs/applications/networking/irc/weechat/scripts/weechat-matrix/default.nix]: Unfortunately one of the shebangs is different to the others, so I think this doesn't quite pass the DRY threshold yet.

**Comment by emilazy** [comment]
pushed with requested tweaks ^^

**Comment by Valodim** [comment]
when I tested this I ran into #79646

**Comment by emilazy** [comment]
As far as I'm aware that's just due to the nature of the nixpkgs WeeChat configuration; you need to ensure WeeChat's Python picks up the necessary package dependencies:

```nix
  environment.systemPackages = [
    ...
    (weechat.override {
      configure = {availablePlugins, ...}: with weechatScripts; {
        plugins = with availablePlugins; [
          (python.withPackages (_: weechat-matrix.propagatedBuildInputs))
        ];
        scripts = [ weechat-matrix ];
      };
    })
    weechatScripts.weechat-matrix
  ];
```

It would be nicer if this happened automatically, but unless I'm mistaken I think it's orthogonal to this update.

**Comment by tilpner** [comment]
`python.withPackages (_: [ weechat-matrix ])` should be sufficient there, no need to explicitly pass `propagatedBuildInputs` around

**Comment by mjlbach** [comment]
@emilazy Is there a reason this is marked as unsupported on macOS? Seems to build fine.

**Comment by jonringer** [comment]
probably a previous issue that blocked mac support. Pretty common

**Comment by emilazy** [comment]
@mjlbach not to my knowledge; feel free to submit a PR that changes it to `platforms.unix` or `platforms.all`, or I'll try to remember to get around to it if I bump the package again.

**Pull request merged**